pkg/k8s/dto.go

```diff
@@ -21,3 +28,42 @@ func DefaultResources(res *sharedAPI.Resources) {
 	res.CPU = "1"
 	res.Memory = "1Gi"
 }
+
+// CheckDeploymentResourcesChanged checks if any of the deployment resources passed by the users have changes in shareAPI.Resources
+func CheckDeploymentResourcesChanged(res *sharedAPI.Resources) bool {
+	if res.CPU != "" ||
+		res.CPULimit != "" ||
+		res.Memory != "" ||
+		res.MemoryLimit != "" ||
+		res.Storage != "" {
+		return true
+	}
+	return false
+}
+
+// DeployReconciliation gets the pod related ot the deployment check if it still stuck in the pending state due to the insufficient resources,
+// then we delete said pod, so it can get created again with updated (decreased) deployment specs resources
+func DeployReconciliation(name string, namespace string) *restErrors.RestErr {
+	var k8Client = NewClientService()
+	// get pod
+	pod := &corev1.Pod{}
+	key := types.NamespacedName{
+		Namespace: namespace,
+		Name:      fmt.Sprintf("%s-0", name),
+	}
+	err := k8Client.Get(context.Background(), key, pod)
+	if apiErrors.IsNotFound(err) {
+		go logger.Error("DEPLOY_RECONCILIATION", err)
+		return restErrors.NewBadRequestError(fmt.Sprintf("pod by name %s doesn't exit", key.Name))
+	}
+
+	//check if pod status
+	if pod.Status.Phase == "Pending" {
+		err = k8Client.Delete(context.Background(), pod)
+		if err != nil {
+			go logger.Error("", err)
+			return restErrors.NewInternalServerError(fmt.Sprintf("can't update deploy by name %s", name))
+		}
+	}
+	return nil
```

Comment:
break this function into:
- getPod(name, ns) -> *pod, err
- GetPodStatus(*pod) -> string
- DeletePod(*pod) -> err


## Very important 🔥

- **BEFORE** updating the actual node, peer ...
  - check if cpu or memory is updated
  - fetch the pod by name: name-0
  - check its status
  - if pod status is pending

- delete the pod **AFTER** updating the node, peer ... with pending pod